vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,372 @@
+			},
+			"telephone": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The user's telephone",
+			},
+			"instant_messaging": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The user's telephone",
+			},
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				ForceNew:    false,
+				Description: "True if the user is enabled and can log in.",
+			},
+			"is_group_role": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    false,
+				Description: "True if this user has a group role.",
+			},
+			"is_locked": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: false,
+				Description: "If the user account has been locked due to too many invalid login attempts, the value " +
+					"will change to true (only the system can lock the user). " +
+					"To unlock the user re-set this flag to false.",
+			},
+			"take_ownership": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    false,
+				Description: "Take ownership of user's objects on deletion.",
+			},
+			"deployed_vm_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     10,
+				ForceNew:    false,
+				Description: "Quota of vApps that this user can deploy. A value of 0 specifies an unlimited quota.",
+			},
+			"stored_vm_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     10,
+				ForceNew:    false,
+				Description: "Quota of vApps that this user can store. A value of 0 specifies an unlimited quota.",
+			},
+		},
+	}
+}
+
+// Converts resource data into a OrgUserConfiguration structure
+// used for both creation and update.
+func resourceToUserData(d *schema.ResourceData, meta interface{}) (*govcd.OrgUserConfiguration, *govcd.AdminOrg, error) {
+	vcdClient := meta.(*VCDClient)
+	orgName := d.Get("org").(string)
+	if orgName == "" {
+		orgName = vcdClient.Org
+	}
+	if orgName == "" {
+		return nil, nil, fmt.Errorf("missing org name")
+	}
+	adminOrg, err := govcd.GetAdminOrgByName(vcdClient.VCDClient, orgName)
+	if err != nil {
+		return nil, nil, err
+	}
+	if adminOrg.AdminOrg == nil || adminOrg.AdminOrg.HREF == "" {
+		return nil, nil, fmt.Errorf("error retrieving org %s", orgName)
+	}
+
+	var userData govcd.OrgUserConfiguration
+	userData.RoleName = d.Get("role").(string)
+
+	userData.Name = d.Get("name").(string)
+	userData.Description = d.Get("description").(string)
+	userData.FullName = d.Get("full_name").(string)
+	userData.EmailAddress = d.Get("email_address").(string)
+	userData.Telephone = d.Get("telephone").(string)
+	userData.ProviderType = d.Get("provider_type").(string)
+	userData.IsEnabled = d.Get("enabled").(bool)
```

Comment:
It would probably be cleaner if we renamed `userData.IsEnabled` to `userData.Enabled` as well to match the adjusted resource field name?

Reply:
you want me to open another PR in govcd for this?
No users of Terraform will see it

Reply:
Forgot that this is in govcd, never mind.